offload-items.lic

```diff
@@ -0,0 +1,159 @@
+
+  @@offload_items_success_patterns = [
+    /^You drop/i,
+    /^You set/i,
+    /^You put/i
+  ]
+
+  # Messages that when trying to put away an item you're warned.
+  # To continue you must retry the command.
+  @@offload_items_retry_patterns = [
+    # You may get the next messages when an outdated item is updated upon use.
+    # "Something appears different about the <item>, perhaps try doing that again."
+    # Example: https://elanthipedia.play.net/Item:Leather_lotion
+    /Something appears different about/,
+    /perhaps try doing that again/
+  ]
+
+  @@offload_items_failure_patterns = [
+    /^You can't put that there/i,
+    /^What were you referring to/i,
+    /^Please rephrase that command/,
+    /^I could not find what you were referring to/,
+    /^You can't put that in|on|under|behind there/i,
+    /^Stow what/,
+    /^Please rephrase that command/,
+    /^What were you referring to/,
+    /^I could not find what you were referring to/,
+    /even after stuffing it/,
+    /is too .* to (fit|hold)/,
+    /close the fan/,
+    /no matter how you arrange it/,
+    /There isn't any more room in/,
+    /There's no room/,
+    /^Weirdly, you can't/,
+    /to fit in the/,
+    /too heavy to go in there/,
+    /You (need to|should) unload/,
+    /You can't do that/,
+    /You just can't get/,
+    /You can't put items/,
+    /You can only take items out/,
+    /Perhaps you should be holding that first/,
+    /Containers can't be placed in/,
+    /The .* is not designed to carry anything/,
+    # You may get the next message if you've been cursed and unable to let go of items.
+    # Find a Cleric to uncurse you.
+    /Oddly, when you attempt to stash it away safely/,
+    /completely full/,
+    /That doesn't belong in there!/,
+    /exerts a steady force preventing/
+  ]
+
+  def offload_items(source, destination, noun)
+    # Pause script and warn that the script is designed to offload items
+    # from your person to somewhere else. Item loss may occur.
+    DRC.message("WARNING: This script is designed to offload items from YOU to somewhere ELSE.")
+    DRC.message("WARNING: It also has the ability to move every item from the source container.")
+    DRC.message("WARNING: Item loss may occur. Use with caution.")
+    DRC.message("WARNING: If you want to transfer items between containers you own, use ;transfer-items instead.")
+    DRC.message("\n")
+    
+    unless noun
+      DRC.message("You are about to move EVERY SINGLE ITEM from from #{source} to #{destination}.")
+      DRC.message("Unpause the script to continue. Otherwise, ;kill the script to stop.")
+      pause_script
+      DRC.message("Are you SURE you want to offload EVERY SINGLE ITEM from YOUR #{source} to #{destination}?")
+      pause_script
+    else
+      DRC.message("You are about to move all #{noun}s from #{source} to #{destination}.")
+      DRC.message("Unpause the script to continue. Otherwise, ;kill the script to stop.")
+      pause_script
+    end
```

Comment:
💡 Suggest moving lines 75-93 into their own method, maybe something like `warn_before_offload`, and call it right before line 20. This way the warning logic isn't intermingled with the item transfer logic.

Reply:
Thanks you. Can you give a hint with pseudo code or otherwise how I can encapsulate this messaging as you suggest? (I'm quite new to coding and very new to Ruby.)

Reply:
Sure, something like this:

_Lines 75-93 get moved into a new method defined in this class:_
```ruby
def warn_before_offload(source, destination, noun)
    # Pause script and warn that the script is designed to offload items
    # from your person to somewhere else. Item loss may occur.
    DRC.message("WARNING: This script is designed to offload items from YOU to somewhere ELSE.")
    DRC.message("WARNING: It also has the ability to move every item from the source container.")
    DRC.message("WARNING: Item loss may occur. Use with caution.")
    DRC.message("WARNING: If you want to transfer items between containers you own, use ;transfer-items instead.")
    DRC.message("\n")

    unless noun
      DRC.message("You are about to move EVERY SINGLE ITEM from from #{source} to #{destination}.")
      DRC.message("Unpause the script to continue. Otherwise, ;kill the script to stop.")
      pause_script
      DRC.message("Are you SURE you want to offload EVERY SINGLE ITEM from YOUR #{source} to #{destination}?")
      pause_script
    else
      DRC.message("You are about to move all #{noun}s from #{source} to #{destination}.")
      DRC.message("Unpause the script to continue. Otherwise, ;kill the script to stop.")
      pause_script
    end
end
```

_Call the new warn method when the class is initialized:_
```ruby
  def initialize
    # ... existing logic ...
    warn_before_offload(args.source, args.destination, args.noun)
    offload_items(args.source, args.destination, args.noun)
  end
```

Reply:
Ah, duh, makes sense now that you outline it. Will probably be tomorrow but I'll edit and commit. Let me know what you think about my comments on `DRCI.put_away_item_unsafe?`. Would have liked to use that helper method but it didn't seem to do what I needed.

Reply:
Broke the warning messaging out per @KatoakDR .